scripts/components/Keyboard.js

```diff
@@ -0,0 +1,135 @@
+import TextArea from './TextArea.js';
+import Button from './Button.js';
+
+let instance = null;
+
+class Keyboard {
+  constructor(keyboardArray, options = {}) {
+    if (instance) {
+      return instance;
+    }
+
+    this.lang = localStorage.getItem('lang') || 'ru';
+    this.ctrlIsPressed = false;
+    this.altIsPressed = false;
+    this.isShiftPressed = false;
+    this.isCapsLockActive = false;
+    this.el = null;
+    this.options = options;
+    this.keyboardArray = keyboardArray;
+    this.buttons = null;
+  }
+
+  generateButtonsArray() {
+    this.buttons = this.keyboardArray.map((item) => {
+      if (item.isOperating) {
+        return new Button(this.handleButtonClick, item.code, item.label, item.additionalClasses, {
```

Comment:
this is readable enough, but you don't necessary need to duplicate this code. Please think of how to refactor it to call `return new Button` only once in this function.

Reply:
I have fixed this case. Do I have to click resolve conversation on my own? Or you will do that after checking?

Reply:
No, this is my job to check & click if fixed.

Reply:
Ok, I will inform you after fixing everything

Reply:
Done